client/internal/engine.go

```diff
@@ -216,7 +216,33 @@ func (e *Engine) Start() error {
 	return nil
 }
 
-// removePeers finds and removes peers that do not exist anymore in the network map received from the Management Service
+// modifyPeers detects peers that have been modified (e.g. IP address has been changed) and simply removes them
+// A consequent call to addPeers will add them again but already with a new configuration
+// Has to be always called before addPeers.
+func (e *Engine) modifyPeers(peersUpdate []*mgmProto.RemotePeerConfig) error {
+	var modified []*mgmProto.RemotePeerConfig
+
+	// first, check if peers have been modified
+	for _, p := range peersUpdate {
+		if peerConn, ok := e.peerConns[p.GetWgPubKey()]; ok {
+			if peerConn.GetConf().ProxyConfig.AllowedIps != strings.Join(p.AllowedIps, ",") {
+				modified = append(modified, p)
+			}
+		}
+	}
+
+	// second, terminate the connection and remove them
+	for _, p := range modified {
+		err := e.removePeer(p.GetWgPubKey())
+		if err != nil {
+			return err
+		}
+	}
```

Comment:
As we discussed, it might make sense to keep the logic to re-add the peer here as well. I would extract the loop code from e.addNewPeers and create a new one for it, maybe e.addNewPeeR to be called by e.addNewPeers and e.modifyPeers